plugin/src/main/scala/se/nullable/sbtix/NixWriter.scala

```diff
@@ -86,17 +86,17 @@ case class NixArtifactCollection(artifacts: Seq[NixArtifact])  extends NixBuilde
 
   def toNixValue: String = 
       s"""{
-          |  ${indent(artifacts.map(_.toNix).mkString("\r\n"))}
+          |  ${indent(artifacts.distinct.map(_.toNix).mkString("\r\n"))}
           |}""".stripMargin
 
 }
 
-case class NixArtifact(repoName:String, relative: String, url: URL, sha256: String) extends NixBuilder {
-  def toNixRef = s"${quote(repoName + "/" + relative.stripPrefix("/"))}"
+case class NixArtifact(repoName:String, relative: String, url: String, sha256: String) extends NixBuilder {
```

Comment:
Yeah, it was being a pain. With URL in the case class the output file had duplicates and Nix wouldn't accept it.  Could wrap the string in its own case class. Or wrap URL and write a hashcode function for it.